egs/wsj/s5/steps/nnet3/compute_and_adjust_priors.py

```diff
@@ -0,0 +1,252 @@
+logger.addHandler(handler)
+logger.info('Starting compute and adjusting posteriors (train_dnn.py)')
+
+
+def get_args():
+    """ Get args from stdin.
+
+    We add compulsary arguments as named arguments for readability
+
+    The common options are defined in the object
+    libs.nnet3.train.common.CommonParser.parser.
+    See steps/libs/nnet3/train/common.py
+    """
+
+    parser = argparse.ArgumentParser(
+        description="""Compute the posteriors using model posteriors computed
+            on example data. It re-adjusts the prior using computed average
+            posterior.
+        """,
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+        conflict_handler='resolve')
+        #parents=[common_train_lib.CommonParser().parser])
+
+    # options for average posterior computation
+    parser.add_argument("--post-process.prior-subset-size", type=int,
+                        dest='prior_subset_size', default=20000,
+                        help="Number of samples for computing priors")
+    parser.add_argument("--post-process.num-jobs-compute-prior", type=int,
+                        dest='num_jobs_compute_prior', default=10,
+                        help="The prior computation jobs are single "
+                        "threaded and run on the CPU")
+    parser.add_argument("--post-process.output-name", type=str,
+                       dest='output_name',
+                       default='output',
+                       help='The average posteriors computed and adjusted '
+                            'for this output node.')
+    parser.add_argument("--post-process.model", type=str,
+                       action=common_lib.NullstrToNoneAction,
+                       dest='model',
+                       default=None,
+                       help='The model name in dir used to compute average posteriors'
+                            ' w.r.t its output node.')
+    parser.add_argument("--post-process.readjust-model", type=str, 
+                        dest='readjusted_model',
+                        default="final",
+                        help="the model name used to store readjusted model.")
+    parser.add_argument("--post-process.readjust-priors", type=str, 
+                        action=common_lib.StrToBoolAction,
+                        dest='readjust_priors',
+                        default=True, choices=["true", "false"],
+                        help='If true, the prior re-adjusted using computed '
+                             'averarage posterior.')
+    # extra egs option
+    parser.add_argument("--egs.use-multitask-egs", type=str, 
+                        action=common_lib.StrToBoolAction,
+                        dest='use_multitask_egs',
+                        default=False, choices=["true", "false"],
+                        help='If true, it is assumed there are different examples '
+                          'correspond to multiple tasks in output layer.')
```

Comment:
The idea is that, it computes average posterior and adjust the prior inside the function, but we need to compute and adjust it per output, so I moved this computation outside train_raw_dnn.py!
I think it shouldn't be really a part of training script.
@vimalmanohar do you have any suggestion??